docs/contributor-guide.rst

```diff
@@ -135,3 +135,22 @@ Editor config
 
 - `python-black <https://atom.io/packages/python-black>`_
 - `linter-flake8 <https://atom.io/packages/linter-flake8>`_
+
+Overview of Resource and Collection interaction
```

Comment:
I think the title should mention something about `from_json` vs `from_data` so that its more easily searchable/findable